src/org/labkey/test/tests/component/EntityInsertPanelTest.java

```diff
@@ -0,0 +1,123 @@
+package org.labkey.test.tests.component;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.TestTimeoutException;
+import org.labkey.test.categories.DailyB;
+import org.labkey.test.components.ui.EntityInsertPanel;
+import org.labkey.test.pages.test.CoreComponentsTestPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.SampleTypeDefinition;
+import org.labkey.test.util.TestDataGenerator;
+import org.labkey.test.util.exp.SampleTypeAPIHelper;
+import org.openqa.selenium.WebElement;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+@Category({DailyB.class})
+public class EntityInsertPanelTest extends BaseWebDriverTest
+{
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        EntityInsertPanelTest init = (EntityInsertPanelTest) getCurrentTest();
+
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+    }
+
+    @Before
+    public void preTest() throws Exception
+    {
+        goToProjectHome();
+    }
+
+    @Test
+    public void testInsertSamplesFromGrid() throws Exception
+    {
+        String sampleTypeName = "insert_from_edit_grid";
+        SampleTypeDefinition props = new SampleTypeDefinition(sampleTypeName).setFields(standardTestSampleFields());
+        TestDataGenerator dgen = SampleTypeAPIHelper.createEmptySampleType(getProjectName(), props);
+        CoreComponentsTestPage testPage = CoreComponentsTestPage.beginAt(this, getProjectName());
+        EntityInsertPanel testPanel = testPage.getEntityInsertPanel();
+        testPanel.getEntityTypeSelect("Sample Type").select(sampleTypeName);
+
+        String pasteText = "ed\tbrother\tthe quiet one\t\t2\tstringfellow\t11/11/2020\ttrue\n" +
+                "jed\tother brother\tthe squinty one\t\t3\tstrongfellow\t11/11/2020\tfalse\n" +
+                "ted\tother other brother\tisn't sure about the others\t\t4\tstrangefellow\t11/11/2020\ttrue";
```

Comment:
I guess I'm still a little confused about this test data. 
It looks like test validates the number of rows added but not the content. If test does not care about the content and only the count, this data is a little bit of overkill.
If there is a concern that the form may not accurately set a field (because this is validating the test form and not LKS), then the validation code below should also validate that the fields have the values that were set.

Reply:
OK, I will validate the complete dataset here before pushing